### PR TITLE
setup system wide proxy

### PR DESCRIPTION
**Pull request by rcerven**
* CLOUDBLD-8735

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough

### CODE REVIEW

**Comment by MartinBasti** [comment]
This should be optional, not all deployments have this requirement.

**Comment by rcerven** [comment]
> This should be optional, not all deployments have this requirement.

added option

**Review by mkosiarc — APPROVED** [review]

File: tasks/main.yaml [hunk 1]
```diff
@@ -58,3 +58,34 @@
...
+  with_items:
+    - {regex: 'export HTTPS_PROXY=.+'}
+    - {regex: 'export NO_PROXY=.+'}
+    - {regex: 'export https_proxy=.+'}
+    - {regex: 'export no_proxy=.+'}
+    - {regex: 'HTTPS_PROXY=.+'}
+    - {regex: 'NO_PROXY=.+'}
+    - {regex: 'https_proxy=.+'}
+    - {regex: 'no_proxy=.+'}
+  when: setup_proxy
```
Comment (chmeliik) [inline tasks/main.yaml:76]: Why do we need this one? Wouldn't the second task that replaces them based on a regex be enough?

Removing the lines first has a minor downside that both this task and the next one will always report `changed: true`

Comment (rcerven) [reply tasks/main.yaml:76]: this is exactly same as we setup osbs clusters

the first task removes duplicate lines, 2nd changes configururation

Comment (chmeliik) [reply tasks/main.yaml:76]: well alright then `¯\_(ツ)_/¯`

**Review by chmeliik — APPROVED** [review]

**Pull request merged**